webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -111,44 +147,39 @@ const ProjectDetails = memo((projectDetails: ProjectDetails) => {
 
     useEffect(() => {
         /**
-         * Prevent calling API to fetch subscription list twice on switching channel
-         *
-         * If the current channel is changed and "showAllSubscriptions" was true on the last channel then
-         * this useEffect runs twice because "subscriptionListApiParams" is modified twice
-         * once when "currentChannelId" is updated and other time when "showAllSubscriptions" is set to false
+         * If all subscriptions for a project are already loaded then do not make API calls on switching channel
          */
-        if (showAllSubscriptions && previousState?.currentChannelId !== currentChannelId) {
+        if (previousState?.currentChannelId !== currentChannelId && showAllSubscriptions) {
             return;
         }
         fetchSubscriptionList();
-    }, [subscriptionListApiParams]);
-
-    useEffect(() => {
-        setShowAllSubscriptions(false);
-    }, [currentChannelId]);
+    }, [subscriptionListApiParams.channel_id, subscriptionListApiParams.project, subscriptionListApiParams.offset]);
 
     // Fetch the subscription list when new subscription is created
     useEffect(() => {
         if (getSubscribeModalState(state).isCreated) {
             dispatch(toggleIsSubscribed(false));
-            fetchSubscriptionList();
+            handlePagination(true);
         }
     }, [getSubscribeModalState(state).isCreated]);
 
-    const {data, isLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName, subscriptionListApiParams);
-    const subscriptionList = data as SubscriptionDetails[];
-
-    const {isLoading: isUnlinkProjectLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails);
-    const {isLoading: isDeleteSubscriptionLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.deleteSubscription.apiServiceName, subscriptionToBeDeleted);
-
     // Update the subscription list on RHS when a subscription is deleted using the slash command
     useEffect(() => {
         if (getWebsocketEventState(state).isSubscriptionDeleted) {
-            fetchSubscriptionList();
+            handlePagination(true);
             dispatch(toggleIsSubscriptionDeleted(false));
         }
     }, [getWebsocketEventState(state).isSubscriptionDeleted]);
 
+    useEffect(() => {
+        if (!showAllSubscriptions && subscriptionList?.length) {
```

Comment:
we are already setting `subscriptionList` empty initially. I don't think there is a need for this.
```suggestion
        if (!showAllSubscriptions && subscriptionList.length) {
```